ir/instr.cpp

```diff
@@ -1495,4 +1495,73 @@ unique_ptr<Instr> Store::dup(const string &suffix) const {
   return make_unique<Store>(*ptr, *val, align);
 }
 
+
+vector<Value*> Memset::operands() const {
+  return { ptr, val, bytes };
+}
+
+void Memset::rauw(const Value &what, Value &with) {
+  RAUW(ptr);
+  RAUW(val);
+  RAUW(bytes);
+}
+
+void Memset::print(ostream &os) const {
+  os << "memset " << *ptr << " align " << align << ", " << *val
+     << ", " << *bytes;
+}
+
+StateValue Memset::toSMT(State &s) const {
+  auto &[vptr, np] = s[*ptr];
+  auto &[vbytes, np2] = s[*bytes];
+  s.addUB(vbytes.ugt(0).implies(np));
+  s.addUB(vbytes.ugt(0).implies(np2));
+  s.getMemory().memset(vptr, s[*val], vbytes, align);
+  return {};
+}
+
+expr Memset::getTypeConstraints(const Function &f) const {
+  return ptr->getType().enforcePtrType() && bytes->getType().enforceIntType();
+}
+
+unique_ptr<Instr> Memset::dup(const string &suffix) const {
+  return make_unique<Memset>(*ptr, *val, *bytes, align);
+}
+
+
+vector<Value*> Memcpy::operands() const {
+  return { dst, src, bytes };
+}
+
+void Memcpy::rauw(const Value &what, Value &with) {
+  RAUW(dst);
+  RAUW(src);
+  RAUW(bytes);
+}
+
+void Memcpy::print(ostream &os) const {
+  os << "memcpy " << *dst  << " align " << align_dst << ", " << *src
+     << " align " << align_src << ", " << *bytes;
+}
+
+StateValue Memcpy::toSMT(State &s) const {
+  auto &[vdst, np] = s[*dst];
+  auto &[vsrc, np2] = s[*src];
+  auto &[vbytes, np3] = s[*bytes];
+  s.addUB(vbytes.ugt(0).implies(np));
+  s.addUB(vbytes.ugt(0).implies(np2));
+  s.addUB(vbytes.ugt(0).implies(np3));
```

Comment:
same comment as for memset: should be addUB(np3) only.
Also, please rename vars to np_dst, np_src, or something like that. np2/np3 are error prone :)